java/Nikiforov-Oleksandr/src/com/zoolatech/lecture3/tasks/_2/Validator.java

```diff
@@ -0,0 +1,44 @@
+package com.zoolatech.lecture3.tasks._2;
+
+public interface Validator {
+    boolean isValid(UserAccount userAccount);
+
+    default String getStringFieldValue(UserAccount userAccount, String fieldName) {
```

Comment:
Even though default methods provide some implementation, they are still a part of an interface. This means that they can be called outside. 

When you define an interface, try to think what behaviour of children classes it should enforce. Should validators, main role of which is to validate data, be responsible for field extraction? 

As for me, it might be better to define a separate class (e.g., UserAccountFieldExtractor) and allow it to deal with this. Then, you can pass an instance of such extractor to specific validators. Such approach will also simplify your testing, since you can isolate field extraction from validation logics.